internal/db/repos_db_test.go

```diff
@@ -1089,6 +1107,681 @@ func TestRepos_List_externalServiceID(t *testing.T) {
+		{"All", ReposListOptions{IDs: all.IDs()}, repoNamesFromRepos(all)},
+		{"Default", ReposListOptions{}, repoNamesFromRepos(all)},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			repos, err := Repos.ListRepoNames(ctx, test.opt)
+			if err != nil {
+				t.Fatal(err)
+			}
+			assertJSONEqual(t, test.want, repos)
+		})
+	}
+}
+
+func TestRepos_ListRepoNames_serviceTypes(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	dbtesting.SetupGlobalTestDB(t)
+	ctx := actor.WithInternalActor(context.Background())
+
+	mine := mustCreate(ctx, t, types.MakeGithubRepo())
+	yours := mustCreate(ctx, t, types.MakeGitlabRepo())
+	others := mustCreate(ctx, t, types.MakeGitoliteRepo())
+	both := append(mine, yours...)
+	all := append(both, others...)
+
+	tests := []struct {
+		name string
+		opt  ReposListOptions
+		want []*types.RepoName
+	}{
+		{"OnlyGithub", ReposListOptions{ServiceTypes: []string{extsvc.TypeGitHub}}, repoNamesFromRepos(mine)},
+		{"OnlyGitlab", ReposListOptions{ServiceTypes: []string{extsvc.TypeGitLab}}, repoNamesFromRepos(yours)},
+		{"Both", ReposListOptions{ServiceTypes: []string{extsvc.TypeGitHub, extsvc.TypeGitLab}}, repoNamesFromRepos(both)},
+		{"Default", ReposListOptions{}, repoNamesFromRepos(all)},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			repos, err := Repos.ListRepoNames(ctx, test.opt)
+			if err != nil {
+				t.Fatal(err)
+			}
+			assertJSONEqual(t, test.want, repos)
+		})
+	}
+}
+
+func TestRepos_ListRepoNames_pagination(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	dbtesting.SetupGlobalTestDB(t)
+	ctx := actor.WithInternalActor(context.Background())
+
+	createdRepos := []*types.Repo{
+		{Name: "r1"},
+		{Name: "r2"},
+		{Name: "r3"},
+	}
+	for _, repo := range createdRepos {
+		mustCreate(ctx, t, repo)
+	}
+
+	type testcase struct {
+		limit  int
+		offset int
+		exp    []api.RepoName
+	}
+	tests := []testcase{
+		{limit: 1, offset: 0, exp: []api.RepoName{"r1"}},
+		{limit: 1, offset: 1, exp: []api.RepoName{"r2"}},
+		{limit: 1, offset: 2, exp: []api.RepoName{"r3"}},
+		{limit: 2, offset: 0, exp: []api.RepoName{"r1", "r2"}},
+		{limit: 2, offset: 2, exp: []api.RepoName{"r3"}},
+		{limit: 3, offset: 0, exp: []api.RepoName{"r1", "r2", "r3"}},
+		{limit: 3, offset: 3, exp: nil},
+		{limit: 4, offset: 0, exp: []api.RepoName{"r1", "r2", "r3"}},
+		{limit: 4, offset: 4, exp: nil},
+	}
+	for _, test := range tests {
+		repos, err := Repos.ListRepoNames(ctx, ReposListOptions{LimitOffset: &LimitOffset{Limit: test.limit, Offset: test.offset}})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if got := sortedRepoNames(reposFromRepoNames(repos)); !reflect.DeepEqual(got, test.exp) {
+			t.Errorf("for test case %v, got %v (want %v)", test, got, test.exp)
+		}
+	}
+}
+
+// TestRepos_ListRepoNames_query tests the behavior of Repos.List when called with
+// a query.
+// Test batch 1 (correct filtering)
+func TestRepos_ListRepoNames_query1(t *testing.T) {
```

Comment:
Why not just name it 
```suggestion
func TestRepos_ListRepoNames_correctFiltering(t *testing.T) {
```